src/dotnet/install.sh

```diff
@@ -53,440 +22,145 @@ err() {
-        DOTNET_DOWNLOAD_VERSION=""
-    fi
-
-    dotnet_patchless_version="$(echo "${DOTNET_DOWNLOAD_VERSION}" | cut -d "." --field=1,2)"
-
-    set +e
-    dotnet_channel_version="$(curl -s "${DOTNET_CDN_FEED_URI}/dotnet/release-metadata/releases-index.json" | jq -r --arg channel_version "${dotnet_patchless_version}" '[."releases-index"[]] | sort_by(."channel-version") | reverse | map( select(."channel-version" | startswith($channel_version))) | first | ."channel-version"')"
-    set -e
-
-    # Construct the releases URL using the official channel-version if one was found.  Otherwise make a best-effort using the user input.
-    if [ -n "${dotnet_channel_version}" ] && [ "${dotnet_channel_version}" != "null" ]; then
-        dotnet_releases_url="${DOTNET_CDN_FEED_URI}/dotnet/release-metadata/${dotnet_channel_version}/releases.json"
+# Installs a version of .NET using the DOTNET_INSTALLER_SCRIPT
+install_version() {
+    local version="$1"
+    local channel="STS"
+
+    echo "Installing version '$version'..."
+
+    # Quick options reminder for dotnet-install.sh:
+    #
+    # --version: 'latest' (default) or an exact version in the form 'A.B.C' like '6.0.412'
+    # --channel: 'LTS' (default), 'STS', a two-part version in the form 'A.B' like '6.0' or three-part form 'A.B.Cxx' like '6.0.1xx'
+    # --quality: 'daily', 'signed', 'validated', 'preview' or 'GA'
+    #
+    # Valid examples
+    #
+    # dotnet-install.sh [--version latest] --channel LTS
+    # dotnet-install.sh [--version latest] --channel STS
+    # dotnet-install.sh [--version latest] --channel 6.0 [--quality GA]
+    # dotnet-install.sh [--version latest] --channel 6.0.4xx [--quality GA]
+    # dotnet-install.sh [--version latest] --channel 8.0 --quality preview
+    # dotnet-install.sh [--version latest] --channel 8.0 --quality daily
+    # dotnet-install.sh --version 6.0.412
+    #
+    # The channel option is only used when version is 'latest' because an exact version overrides the channel option
+    # The quality option is only used when channel is 'A.B' or 'A.B.Cxx' because it can't be used with STS or LTS
+    #
+    # This script aims to reduce these combinations of options to a single 'version' input
+    # Currently this script does not make it possible to request a version in the form 'A.B' or 'A.B.Cxx' and a quality other than 'GA'
+    if [[ "$version" == "latest" ]]; then
+        # Fetch the latest version manually, because dotnet-install.sh does not support it directly
+        version=$(fetch_latest_sdk_version)
+        channel=""
+    elif [[ "$version" == "lts" ]]; then
+        # When user input is 'lts'
+        # Then version=latest, channel=LTS
+        channel="LTS"
+        version="latest"
+    elif [[ "$version" =~ ^[0-9]+\.[0-9]+$ ]]; then
+        # When user input is form 'A.B' like '3.1'
+        # Then version=latest, channel=3.1
+        channel="$version"
+        version="latest"
+    elif [[ "$version" =~ ^[0-9]+\.[0-9]+\.[0-9]xx$ ]]; then
```

Comment:
From what I understand, these numbers are incremented with Visual Studio service releases. So for 7.0.4xx you need to update your VS2022 when coming from 7.0.3xx or lower.

I am not sure if this is at all relevant to someone using VS Code in a Linux container, so I am okay with removing this case. I am not entirely sure why the option exists, that's why I included it as a safe default.

Reply:
@bderusha I identified one area where this does matter: `dotnet test` uses [VSTest](https://github.com/microsoft/vstest) and updating it may cause issues like https://github.com/microsoft/vstest/issues/4657. So it can be favorable to stay on an older feature band like 7.0.3xx without pinning to an exact patch version.

I think that justifies keeping it and marking as resolved.